File: src/main/java/br/com/alisonviana/service/StudentService.java
package br.com.alisonviana.service;

import br.com.alisonviana.entity.Student;

import java.util.List;

public interface StudentService {
    List<Student> getAllStudents();
    Student getStudentById(Long id);
    Student saveStudent(Student student);
    Student updateStudent(Student student);
    void deleteStudentById(Long id);
}

File: src/main/kotlin/br/com/alisonviana/service/StudentService.kt
package br.com.alisonviana.service

import br.com.alisonviana.entity.Student
import org.springframework.stereotype.Service

interface StudentService {
    fun getAllStudents(): List<Student>
    fun getStudentById(id: Long): Student
    fun saveStudent(student: Student): Student
    fun updateStudent(student: Student): Student
    fun deleteStudentById(id: Long)
}